### PR TITLE
Remove trailing semicolon on single line html style attribute

### PR DESCRIPTION
**Pull request by bschlenk**
If an inline style attribute doesn't break, then omit the trailing semicolon.

Closes https://github.com/prettier/prettier/issues/7907

- [x] I’ve added tests to confirm my change works.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-css/printer-postcss.js [hunk 1]
```diff
@@ -182,7 +182,7 @@ function genericPrint(path, options, print) {
             !parentNode.raws.semicolon &&
             options.originalText[options.locEnd(node) - 1] !== ";"
           ? ""
-          : ";",
+          : ifBreak(";", ""),
```
Comment (fisker) [inline src/language-css/printer-postcss.js:185]:
```suggestion
          : options.__isHTMLStyleAttribute ? ifBreak(";", "") : ";", 
```

File: tests/html_attributes/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -559,7 +558,7 @@ style="css-prop-1: css-value;css-prop-2: css-value;css-prop-3: css-value;css-pro
   "
 ></div>
 
-<div style="color: red; display: inline;"></div>
+<div style="color: red display: inline"></div>
```
Comment (fisker) [inline tests/html_attributes/__snapshots__/jsfmt.spec.js.snap:561]: Broken

File: src/language-css/printer-postcss.js [hunk 3]
```diff
@@ -182,6 +184,8 @@ function genericPrint(path, options, print) {
             !parentNode.raws.semicolon &&
             options.originalText[options.locEnd(node) - 1] !== ";"
           ? ""
+          : options.__isHTMLStyleAttribute && isLastNode
```
Comment (fisker) [inline src/language-css/printer-postcss.js:187]: https://github.com/bschlenk/prettier/blob/6c8f642ec20693d5a23e97e4b36ddfc56d139a09/src/language-css/printer-postcss.js#L43

We already has this util

**Comment by bschlenk** [comment]
One check was cancelled, what does that mean?

**Comment by fisker** [comment]
@bschlenk I've re-run the check, all good.

**Comment by thorn0** [comment]
@fisker The merge conflict in `tests/html_angular/__snapshots__/jsfmt.spec.js.snap` looks very strange. What caused it?

**Comment by fisker** [comment]
Thank you!

**Comment by paladin2005** [comment]
How Can I disable this rule?

**Comment by evilebottnawi** [comment]
@paladin2005 we don't have rules, prettier is not eslint, it is default behavior and can't be disabled

**Comment by evilebottnawi** [comment]
@fisker what do you think about supporting `--no-semi` in this case?

**Comment by paladin2005** [comment]
Is entering in conflict with Eslint... reverting back to 2.0.5

**Comment by evilebottnawi** [comment]
@paladin2005 What?

**Comment by paladin2005** [comment]
Sorry just ignore or delete my comments. I just restart vscode and everything is fine now.

**Comment by fisker** [comment]
@evilebottnawi  Let's see if other people dislike it.

**Comment by pablomaurer** [comment]
I don't like it =)

**Comment by fisker** [comment]
Can you open a new issue, so we can discuss this?

**Pull request merged**